internal/sops/gcpkms/keysource_test.go

```diff
@@ -0,0 +1,82 @@
+package gcpkms
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestMasterKey_EncryptedDataKey(t *testing.T) {
+	g := NewWithT(t)
+	key := MasterKey{EncryptedKey: "encrypted key"}
+	g.Expect(key.EncryptedDataKey()).To(BeEquivalentTo(string(key.EncryptedKey)))
+}
+
+func TestMasterKey_SetEncryptedDataKey(t *testing.T) {
+	g := NewWithT(t)
+	enc := "encrypted key"
+	key := &MasterKey{}
+	key.SetEncryptedDataKey([]byte(enc))
+	g.Expect(key.EncryptedDataKey()).To(BeEquivalentTo(enc))
+}
+
+func TestMasterKey_EncryptIfNeeded(t *testing.T) {
+	g := NewWithT(t)
+	key := MasterKey{EncryptedKey: "encrypted key"}
+	g.Expect(key.EncryptedDataKey()).To(BeEquivalentTo(string(key.EncryptedKey)))
+
+	key.EncryptIfNeeded([]byte("sops data key"))
+	g.Expect(key.EncryptedDataKey()).To(BeEquivalentTo(string(key.EncryptedKey)))
+}
+
+func TestMasterKey_ToString(t *testing.T) {
+	rsrcId := "/projects/test-flux/locations/global/keyRings/test-flux/cryptoKeys/sops"
+	g := NewWithT(t)
+	key := MasterKeyFromResourceID(rsrcId)
+	g.Expect(key.ToString()).To(Equal(rsrcId))
+}
+
+func TestMasterKey_ToMap(t *testing.T) {
+	g := NewWithT(t)
+	key := MasterKey{
+		credentialJSON: []byte("sensitive creds"),
+		CreationDate:   time.Date(2016, time.October, 31, 10, 0, 0, 0, time.UTC),
+		ResourceID:     "/projects/test-flux/locations/global/keyRings/test/cryptoKeys/sops",
+		EncryptedKey:   "this is encrypted",
+	}
+	g.Expect(key.ToMap()).To(Equal(map[string]interface{}{
+		"resource_id": "/projects/test-flux/locations/global/keyRings/test/cryptoKeys/sops",
+		"enc":         "this is encrypted",
+		"created_at":  "2016-10-31T10:00:00Z",
+	}))
+}
+
+func TestMasterKey_createCloudKMSService(t *testing.T) {
```

Comment:
You can quite easily mock the Google Cloud API for testing purposes to add further tests, [this is also done in the source-controller](https://github.com/fluxcd/source-controller/blob/main/pkg/gcp/gcp_test.go#L78).